consensus/istanbul/validator/weighted.go

```diff
@@ -585,17 +622,37 @@ func (valSet *weightedCouncil) Refresh(hash common.Hash, blockNum uint64) error
 	}
 
 	newStakingInfo := reward.GetStakingInfo(blockNum + 1)
-
-	valSet.stakingInfo = newStakingInfo
-	if valSet.stakingInfo == nil {
+	if newStakingInfo == nil {
 		// Just return without updating proposer
 		return errors.New("skip refreshing proposers due to no staking info")
 	}
+	valSet.stakingInfo = newStakingInfo
 
-	weightedValidators, stakingAmounts, err := valSet.getStakingAmountsOfValidators(newStakingInfo)
+	blockNumBig := new(big.Int).SetUint64(blockNum)
+	chainRules := config.Rules(blockNumBig)
+
+	candidates := append(valSet.validators, valSet.demotedValidators...)
+	weightedValidators, stakingAmounts, err := getStakingAmountsOfValidators(candidates, newStakingInfo)
 	if err != nil {
 		return err
 	}
+
+	if chainRules.IsIstanbul {
```

Comment:
```suggestion
	if config.IsIstanbul(new(big.Int).SetUint64(blockNum)) 
```
What about changing like this? 631, 632 lines can be removed.

Reply:
I will update this on the next PR.